universal-login-relayer/test/testhelpers/setupWalletService.ts

```diff
@@ -24,14 +29,28 @@ export default async function setupWalletService(wallet: Wallet) {
   const walletService = new WalletDeploymentService(config as any, ensService, hooks, walletDeployer, fakeBalanceChecker as any, fakeDevicesService as any);
   const callback = sinon.spy();
   hooks.addListener('created', callback);
-  return {provider, wallet, walletService, callback, factoryContract, ensService, fakeDevicesService};
+  return {provider, wallet, walletService, callback, factoryContract, ensService, fakeDevicesService, ensRegistrar, walletContractAddress};
 }
 
-export const createFutureWallet = async (keyPair: KeyPair, ensName: string, factoryContract: Contract, wallet: Wallet, ensService: ENSService) => {
-  const futureContractAddress = computeCounterfactualAddress(factoryContract.address, keyPair.publicKey, await factoryContract.initCode());
+export const getSetupData = async (keyPair: KeyPair, ensName: string, ensService: ENSService, gasPrice: string, relayerAddress: string, ensRegistrarAddress: string, gasToken = ETHER_NATIVE_TOKEN.address) => {
+  const args = await ensService.argsFor(ensName);
+  const deployment = {
+    owners: [keyPair.publicKey],
+    requiredConfirmations: 1,
+    deploymentCallAddress: ensRegistrarAddress,
+    deploymentCallData: new utils.Interface(gnosisSafe.ENSRegistrar.interface as any).functions.register.encode(args),
+    fallbackHandler: AddressZero,
+    paymentToken: gasToken,
+    payment: utils.bigNumberify(gasPrice).mul(DEPLOY_GAS_LIMIT).toString(),
+    refundReceiver: relayerAddress,
+  };
+  return encodeDataForSetup(deployment);
+};
```

Comment:
extract getDeploymentMessage, we have similar variables in other tests

Reply:
Good idea. We can make two deployments: one for simple deployment and the second one for deployment with ENS registration.